Upgrade Spring (REPO-1295), Quartz (REPO-3049) and remove Hibernate (REPO-2873)

killerboot opened the pull request:
* The classes connected to ibatis/dialect were moved to alfresco-repository.
* Major refactoring was done to TransactionSupportUtil class. It includes change of txn resource management. The new Spring 5 code cleans transaction synchronization once the txn is committed/rolled back. This means that "aftercompletion" listeners cannot manage txn resources they saved in earlier stages. The listeners can be caches, validors, locks, etc. In order to overcome the new behaviour a new ResourceHolder was added. The principle is very much similar to Spring internal resource holder. The object holds the current txn resources in ThreadLocal variable. If the txn is suspended another object is created and linked to previous resource holder so that once the txn is restored all of the resources are still in place. The resource holder is cleaned once all of the "aftercompletion" txn listeners have finished their work.
All the tests for this class are part of repository project.

killerboot committed:
REPO-1295 Upgrade Spring, quartz

killerboot committed:
REPO-2873 Remove Hibernate, move Dialect detection to repository

killerboot committed:
REPO-1295 Move ibatis specific classes to repository

killerboot committed:
REPO-1295 Move ibatis resource loader to repository

killerboot committed:
REPO-1295 Upgrade log4j to 2.10.0

killerboot committed:
Revert "REPO-1295 Upgrade log4j to 2.10.0"

This reverts commit 34e06d2a94969028592847962548876b7609cc6d.

killerboot committed:
REPO-1295 Change the transactional listener code (WIP)

alfresco-build committed:
[maven-release-plugin] prepare release spring5-rc1

alfresco-build committed:
[maven-release-plugin] prepare for next development iteration

killerboot committed:
REPO-1295 Use threadlocal to track txn resources within TransactionSupportUtil

alfresco-build committed:
[maven-release-plugin] prepare release spring5-rc2

alfresco-build committed:
[maven-release-plugin] prepare for next development iteration

killerboot committed:
REPO-1295 Cleanup unused code

killerboot committed:
REPO-1295 Bring back beforeCompletion()

killerboot committed:
REPO-1295 Save txn resources if transaction is suspended

alfresco-build committed:
[maven-release-plugin] prepare release spring5-rc3

alfresco-build committed:
[maven-release-plugin] prepare for next development iteration

killerboot committed:
REPO-1295 Remove broken links

killerboot committed:
REPO-1295 Refactor txn resource management

alfresco-build committed:
[maven-release-plugin] prepare release spring5-rc4

alfresco-build committed:
[maven-release-plugin] prepare for next development iteration

killerboot committed:
REPO-1295 Fix resume of transactions

alfresco-build committed:
[maven-release-plugin] prepare release spring5-rc5

alfresco-build committed:
[maven-release-plugin] prepare for next development iteration

alfresco-build committed:
[maven-release-plugin] prepare release spring5-rc6

alfresco-build committed:
[maven-release-plugin] prepare for next development iteration

andrei-rebegea commented on src/main/java/org/alfresco/util/transaction/TransactionSupportUtil.java:
```diff
@@ -55,14 +54,17 @@
     /**
      * The order of synchronization set to be 100 less than the Hibernate synchronization order
```
not sure if the comment is still valid now

killerboot committed:
REPO-1295 Refactor txn resources to use linked resource holders

alfresco-build committed:
[maven-release-plugin] prepare release spring5-rc7

alfresco-build committed:
[maven-release-plugin] prepare for next development iteration

killerboot committed:
REPO-1295 Upgrade Spring to 5.0.4

killerboot committed:
REPO-1295 Fix cleanup of unused resource holder during txn resume

alfresco-build committed:
[maven-release-plugin] prepare release spring5-rc8

alfresco-build committed:
[maven-release-plugin] prepare for next development iteration

killerboot committed:
REPO-1295 Move transaction and cache packages to repository

alfresco-build committed:
[maven-release-plugin] prepare release spring5-rc9

alfresco-build committed:
[maven-release-plugin] prepare for next development iteration

killerboot committed:
Revert "REPO-1295 Move transaction and cache packages to repository"

This reverts commit c57de394f6ddf679c938e31ad950be2a92ce6da2.
The changes break data model compilation.

killerboot committed:
Merge branch 'master' into feature/repo-1295-spring-upgrade

killerboot committed:
REPO-1295 Remove wrong comment

derekhulley commented on src/main/java/org/alfresco/util/transaction/TransactionSupportUtil.java:
```diff
@@ -130,117 +112,131 @@ public static boolean isActualTransactionActive()
-        // get the resource
-        Object resource = txnSynch.resources.get(key);
-        // done
+        if (TransactionSynchronizationManager.isSynchronizationActive())
```
Please add a comment on why this block of code is necessary.

killerboot replied:
Good point. My first thought was that this check is not necessary, but I soon realized that it is not true.
The comments were added.

derekhulley committed:
Removed unnecessary (and half-implemented) synchronization in
TransactionSynchronizationImpl.  Not related to changes but found during
review.

derekhulley reviewed (approved):
Some of the side effect code could do with commenting.
I added an additional commit that was not directly related.

killerboot committed:
REPO-1295 Remove possibly unnecessary check

killerboot committed:
Revert "REPO-1295 Remove possibly unnecessary check"

This reverts commit 0f1ef491e1d1384e66ee512c0fa26e390567d76e.

killerboot committed:
REPO-1295 Add comments, fix variable names
